### DISCUSSION TITLE
Does PapermillOperator provide action to pull data ?

### DISCUSSION THREAD

**Discussion by silenceliang** [apache/airflow#18134] [Q&A] [answered]
Hello, thanks for your reading first.
 I am wondering if `PapermillOperator` from `airflow.providers.papermill.operators.papermill` is able to pull data from my own jupyter-notebook ?
I thought what we can do is to send a few parameters as arguments of PapermillOperator, however, I would like to obtain the signal or result after my jupyter-notebook was done.
If sending data into jupyter-notebook is called push, the pull action is supposed to receive data from it.
Currently, my only way is to generate the output and save it as a file which is going to be accessed in the following task.

I appreciate your response !

**Comment by josh-fell** [comment, marked as answer]
The `PapermillOperator` does have an optional arg for `output_nb` which is the path to a notebook that will contain the result from the notebook being executed.  Since Jupyter Notebooks are really just JSON files with an `.ipynb` extension, you can read the contents of the output notebook and parse the JSON or convert the output notebook to other formats with something like [nbconvert](https://nbconvert.readthedocs.io/en/latest/nbconvert_library.html#Quick-overview).

Perhaps that's what you meant by "generate the output and save it as a file" as your current approach?

**Comment by silenceliang** [comment]
Dear Josh,

Thanks for your quick reply!

Perhaps that's what you meant by "generate the output and save it as a
> file"?

: Yes, that's what I meant to.

Fortunately, that exactly solved my problem by JSON parser.

Have a good day !


Sincerely,
Brian

Josh Fell ***@***.***> 於 2021年9月10日 週五 下午10:10寫道：
